pogom/app.py

```diff
@@ -183,43 +185,132 @@ def set_location(self, location):
+                args.rarity_update_frequency > 0,
+            'rarityFileName': args.rarity_filename,
+            'pokemonCries': not args.no_pokemon and args.pokemon_cries,
             'gyms': not args.no_gyms,
-            'gym_sidebar': not args.no_gym_sidebar,
+            'gymSidebar': (not args.no_gyms or not args.no_raids) and
+                not args.no_gym_sidebar,
+            'gymFilters': not args.no_gyms and not args.no_gym_filters,
             'raids': not args.no_raids,
+            'raidFilters': not args.no_raids and not args.no_raid_filters,
             'pokestops': not args.no_pokestops,
-            'quests': not args.no_quests,
-            'medalpokemon': args.medalpokemon,
-            'ex_parks': args.ex_parks,
-            'nest_parks': args.nest_parks,
-            'rarity': args.rarity_update_frequency > 0,
-            'custom_css': args.custom_css,
-            'custom_js': args.custom_js
+            'quests': not args.no_pokestops and not args.no_quests,
+            'invasions': not args.no_pokestops and not args.no_invasions,
+            'lures': not args.no_pokestops and not args.no_lures,
+            'weather': not args.no_weather,
+            'spawnpoints': not args.no_spawnpoints,
+            'scannedLocs': not args.no_scanned_locs,
+            's2Cells': not args.no_s2_cells,
+            'ranges': not args.no_ranges,
+            'nestParks': args.nest_parks,
+            'nestParksFileName': args.nest_parks_filename,
+            'exParks': args.ex_parks,
+            'exParksFileName': args.ex_parks_filename
         }
 
         return render_template(
             'map.html',
-            lat=self.location[0],
-            lng=self.location[1],
-            showAllZoomLevel=args.show_all_zoom_level,
             lang=args.locale,
-            mapTitle=args.map_title,
-            headerImage=args.header_image,
-            madminUrl=args.madmin_url,
-            donateUrl=args.donate_url,
-            patreonUrl=args.patreon_url,
-            discordUrl=args.discord_url,
-            messengerUrl=args.messenger_url,
-            telegramUrl=args.telegram_url,
-            whatsappUrl=args.whatsapp_url,
-            show=visibility_flags,
-            generateImages=str(args.generate_images).lower(),
-            rarityFileName=args.rarity_filename,
-            exParksFileName=args.ex_parks_filename,
-            nestParksFileName=args.nest_parks_filename)
+            map_title=args.map_title,
+            header_image=not args.no_header_image,
+            header_image_name=args.header_image,
+            madmin_url=args.madmin_url,
+            donate_url=args.donate_url,
+            patreon_url=args.patreon_url,
+            discord_url=args.discord_url,
+            messenger_url=args.messenger_url,
+            telegram_url=args.telegram_url,
+            whatsapp_url=args.whatsapp_url,
+            settings=settings,
+            donate_text=i8ln('Donate'),
```

Comment:
Wouldn't it be better to do all the translations directly in the template? Because we have a lot of them and it clutter this already big file.

Reply:
Good point, that's indeed cleaner. Apparently you can pass python methods to jinja templates, so that's what I've done: https://github.com/cecpk/RocketMAD/commit/1c2df79b204ed6c0b2ee7def03842eceb934b8e1